### ISSUE TITLE
Github crash causes Unity to no longer load project

### ISSUE THREAD

**Issue by softsound** [github-for-unity/Unity#988]
### Prerequisites

Here is my log file:
[github-unity.log](https://github.com/github-for-unity/Unity/files/2697657/github-unity.log)

### Description

I believe the issue is related to Git refusing to merge unrelated histories. I am using Unity 2018.3.0f2 on Windows 10.

(I'm still new to git and I use unity hub (currently 1.4.0)  as well which could play a part)

I created a commit in github (in unity) before I had a repro attached, I did not realize this as I went about setting up the repro. I thought my repro was synced so on my second commit I tried to push. I was not sure if it went through but I didn't see anything saying it didn't so I tried to push a third commit. I got a warning message you can see in the log. This then crashed Unity and Unity would no longer get past the loading screen before crashing from github. So I could no longer open that project. I can successfully open a clean project without github and experience no crashing. Apparently, this project is crashing so hard Unity Hub can't find it in its history so I have to manually reopen it (though it just crashes so it's not all that useful to try and reopen it)

The second commit was just committing the git files and the third commit was adding a cube with a material to a scene. So nothing fancy.

### Steps to Reproduce

1. Download github plug-in
2. Create a commit
3. Login in and setup repro
4. Add some commits, try to push the 2nd and 3rd. (Without pushing the 1st commit)
5. It will error out with the message shown in my log then crash.

**Expected behavior:** I expected to be required to have a repro setup before I could start committing and be signed in first or at least a warning. I also expected some sort of warning a commit failed to push. I hoped for some sort of warning about the initial commits not matching, but none was given.

**Actual behavior:** I committed, signed in, then setup a repro and linked it to my github account. I assumed it was set up right and started to push my commits. My pushed commits failed and I crashed Unity. I could no longer can past the loading screen when I tried to run the project.

**Reproduces how often:** Well the crash is 100%. I could no longer can open this project do to my github history not matching (I assume)

### Additional Information

I was using Unity Hub 1.4.0 to load the project.
I was using Unity 2018.3.0f2.
I was using Github 1.2.0

Some things about Unity Hub:
Removing all assets in the assets folder will have the project show up with the name .git in unity hub. 

But at least it is showing again, it will crash again if you try to load it though.

Removing the .git hidden folder will allow you to reload the now empty project.

Though it will return a pathing issue with the removal of the hidden git folder and require you to reload the project but the project name will be correct again.

**Comment by StanleyGoldman** [comment]
Okay geez, your issue really fell into the void. I apologize about that.

**Comment by StanleyGoldman** [comment]
So looking at your steps...

>     1. Download github plug-in
>     2. Create a commit
>     3. Login in and setup repro
>     4. Add some commits, try to push the 2nd and 3rd. (Without pushing the 1st commit)
>     5. It will error out with the message shown in my log then crash.

Can you elaborate step 3?
Do you mean you logged into the plugin, or that you logged into github.com?

Also, I realize i'm two months late.
Sorry

**Comment by softsound** [comment]
I logged into both. The plugin and the web version (just to double check my commits were being pushed). I think the problem was the order, I made an init commit before I realized I wasn't signed into the plugin correctly so I set up the repro then. But I'm new to git and the plug-in so I messed up that order. I also pushed the commits in the wrong order I didn't realize that would mess it up at the time. I tried to repro it later and the error message came up about committing it in the wrong order came up correctly and it didn't crash so I wasn't able to reproduce this bug later.

Even though I wasn't able to repro bug this later but I worried about messing up again and creating a project I could no longer open so, in the end, I decided to just use desktop Github and so far it's been working great and I was successfully able to push my commits without any issues or pathing issues in unity hub. 

Maybe including a tutorial for future users might help?
